File: BW8AGU/VB/Ch08/BW8AGU.Samples.VB/BW8AGU.Samples.VB/Common/AzureMobileServicesHelper.vb
Imports Microsoft.WindowsAzure.MobileServices

Public NotInheritable Class AzureMobileServicesHelper
    Private Sub New()
    End Sub
    Public Shared Async Sub InsertDataFromAzureMobileSvc(item As SimpleAppointment)
        Await App.bw8agu_mobileserviceClient.GetTable(Of SimpleAppointment)().InsertAsync(item)
    End Sub

    Public Shared Async Function GetDataFromAzureMobileSvc() As Task(Of MobileServiceCollection(Of SimpleAppointment, SimpleAppointment))
        Try
            Dim table As IMobileServiceTable(Of SimpleAppointment) = App.bw8agu_mobileserviceClient.GetTable(Of SimpleAppointment)()

            Return Await table.ToCollectionAsync()
        Catch ex As Exception
            Throw ex
        End Try
    End Function

    Public Shared Async Sub UpdateDataFromAzureMobileSvc(item As SimpleAppointment)
        Try
            Dim table As IMobileServiceTable(Of SimpleAppointment) =
                App.bw8agu_mobileserviceClient.GetTable(Of SimpleAppointment)()

            Await table.UpdateAsync(item)
        Catch ex As Exception
            Throw ex
        End Try
    End Sub
End Class

File: BW8AGU/C#/Ch08/BW8AGU.Samples.CSharp/Common/AzureMobileServicesHelper.cs
using BW8AGU.Samples.CSharp.DataModel;
using Microsoft.WindowsAzure.MobileServices;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BW8AGU.Samples.CSharp.Common
{
    public static class AzureMobileServicesHelper
    {
        public async static void InsertDataFromAzureMobileSvc(SimpleAppointment item)
        {
            await App.bw8agu_mobileserviceClient.GetTable<SimpleAppointment>()
                .InsertAsync(item);
        }

        public async static Task<MobileServiceCollection<SimpleAppointment, SimpleAppointment>> GetDataFromAzureMobileSvc()
        {
            try
            {
                IMobileServiceTable<SimpleAppointment> table =
                    App.bw8agu_mobileserviceClient
                    .GetTable<SimpleAppointment>();

                return await table.ToCollectionAsync<SimpleAppointment>();
            }
            catch (Exception ex)
            { throw ex; }
        }

        public static void UpdateDataFromAzureMobileSvc(SimpleAppointment item)
        {
            try
            {
                IMobileServiceTable<SimpleAppointment> table = App.bw8agu_mobileserviceClient.GetTable<SimpleAppointment>();

                table.UpdateAsync(item);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
    }
}
